xds/internal/client/cds_test.go

```diff
@@ -620,74 +652,85 @@ func (s) TestUnmarshalCluster(t *testing.T) {
 					TypeUrl: version.V3ClusterURL,
 					Value: func() []byte {
 						cl := &v3clusterpb.Cluster{
+							Name:                 "test",
 							ClusterDiscoveryType: &v3clusterpb.Cluster_Type{Type: v3clusterpb.Cluster_STATIC},
 						}
 						mcl, _ := proto.Marshal(cl)
 						return mcl
 					}(),
 				},
 			},
+			wantUpdate: map[string]ClusterUpdate{"test": {}},
+			wantMD: UpdateMetadata{
+				Status:  ServiceStatusNACKed,
+				Version: testVersion,
+				ErrState: &UpdateErrorMetadata{
+					Version: testVersion,
+					Err:     errPlaceHolder,
+				},
+			},
 			wantErr: true,
 		},
 		{
-			name: "v2 cluster",
-			resources: []*anypb.Any{
-				{
-					TypeUrl: version.V3ClusterURL,
-					Value: func() []byte {
-						mcl, _ := proto.Marshal(v2Cluster)
-						return mcl
-					}(),
+			name:      "v2 cluster",
+			resources: []*anypb.Any{v2ClusterAny},
+			wantUpdate: map[string]ClusterUpdate{
+				v2ClusterName: {
+					ServiceName: v2Service, EnableLRS: true,
+					Raw: v2ClusterAny,
 				},
 			},
-			wantUpdate: map[string]ClusterUpdate{
-				v2ClusterName: {ServiceName: v2Service, EnableLRS: true},
+			wantMD: UpdateMetadata{
+				Status:  ServiceStatusACKed,
+				Version: testVersion,
 			},
 		},
 		{
-			name: "v3 cluster",
-			resources: []*anypb.Any{
-				{
-					TypeUrl: version.V3ClusterURL,
-					Value: func() []byte {
-						mcl, _ := proto.Marshal(v3Cluster)
-						return mcl
-					}(),
+			name:      "v3 cluster",
+			resources: []*anypb.Any{v3ClusterAny},
+			wantUpdate: map[string]ClusterUpdate{
+				v3ClusterName: {
+					ServiceName: v3Service, EnableLRS: true,
+					Raw: v3ClusterAny,
 				},
 			},
-			wantUpdate: map[string]ClusterUpdate{
-				v3ClusterName: {ServiceName: v3Service, EnableLRS: true},
+			wantMD: UpdateMetadata{
+				Status:  ServiceStatusACKed,
+				Version: testVersion,
 			},
 		},
 		{
-			name: "multiple clusters",
-			resources: []*anypb.Any{
-				{
-					TypeUrl: version.V3ClusterURL,
-					Value: func() []byte {
-						mcl, _ := proto.Marshal(v2Cluster)
-						return mcl
-					}(),
+			name:      "multiple clusters",
+			resources: []*anypb.Any{v2ClusterAny, v3ClusterAny},
+			wantUpdate: map[string]ClusterUpdate{
+				v2ClusterName: {
+					ServiceName: v2Service, EnableLRS: true,
+					Raw: v2ClusterAny,
 				},
-				{
-					TypeUrl: version.V3ClusterURL,
-					Value: func() []byte {
-						mcl, _ := proto.Marshal(v3Cluster)
-						return mcl
-					}(),
+				v3ClusterName: {
+					ServiceName: v3Service, EnableLRS: true,
+					Raw: v3ClusterAny,
 				},
 			},
-			wantUpdate: map[string]ClusterUpdate{
-				v2ClusterName: {ServiceName: v2Service, EnableLRS: true},
-				v3ClusterName: {ServiceName: v3Service, EnableLRS: true},
+			wantMD: UpdateMetadata{
+				Status:  ServiceStatusACKed,
+				Version: testVersion,
```

Comment:
Can we also have a case where there are two resources where one of them is good, but the other is bad.

Same for other resource tests too.